dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java

```diff
@@ -24,21 +24,25 @@
 @ParametersAreNonnullByDefault
 final class SnowflakeTaskUtil {
 
+  private static final String EMPTY_WHERE_CLAUSE = "";
+
   static AbstractJdbcTask<Summary> withFilter(
       String format,
       String schemaName,
       String zipEntryName,
       String whereClause,
       Class<? extends Enum<?>> header) {
+    if (format.contains(" WHERE ")) {
+      whereClause = EMPTY_WHERE_CLAUSE;
```

Comment:
Will it construct a valid sql? Do we have test for it? I see the files with sql below, but which line from those should verify this `if` statement is unclear. 

I'm interested how the empty string will be used in sql query. 
`select * from table where 'EMPTY_WHERE_CLAUSE' order by 1;` is not a valid sql.

Reply:
This only works because we have full control of the SQL statements and we know there is no SQL containing clauses after `WHERE` and that `format` will only contain a `WHERE` clause if that is specified with an override. In case of `query` override this code is not executed (i.e. the query is used literally).

This fix would've been more evident if this PR came in two steps, as without this change the queries based on `ACCOUNT_USAGE` would all end in `WHERE DELETES IS NOT NULL WHERE SQL_OVERRIDE(...)`.

The purpose of this PR is to set a test to support refactoring of the query building methods.

Reply:
Can we maybe write a test that will fail if some sql queries with clauses after `WHERE` will appear in the future? 

I'm worried about maintainability. This statement might be confusing for other (including external) developers without additional explanation as in the comment above.

Reply:
I updated the PR to not introduce fixes and only add a test to assert the SQL statements produced by the Snowflake connector. Fixes for illegal SQL will be included in the next PR.